### PR TITLE
Some improvements to the ch4:ofi IOV code

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

The CH4 OFI IOV code is quite messy.  This PR simplifies some parts of it, though not all.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch4/most

**Comment by raffenet** [comment]
Should I ignore the test failures, or are those still being worked on? I'm happy to see so much of this code deleted, tbh.

**Comment by pavanbalaji** [comment]
No, I'll look at the failure.

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch4/most

File: src/mpid/ch4/netmod/ofi/ofi_pre.h [hunk 1]
```diff
@@ -127,20 +127,18 @@ typedef struct {
...
     int util_id;
     MPI_Datatype datatype;
     union {
-        MPIDI_OFI_pack_t *pack;
+        struct {
+            void *buf;
+            size_t count;
+            MPI_Datatype datatype;
+            char *pack_buffer;
+        } pack;
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_pre.h:136]: This grows the request structure size by 16 bytes on x86, if I understand correctly. In the past, there was concern about going beyond a cacheline boundary. I can do a quick experiment to verify if we are within bounds.

Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/ofi_pre.h:136]: Current size of `MPIR_Request` is 324 bytes with `ch4/ofi`, so this increase is no problem.

Comment (pavanbalaji) [reply src/mpid/ch4/netmod/ofi/ofi_pre.h:136]: FWIW, the request allocation itself is dynamic (mostly), so the cache size makes little difference here.  In `ch3`, we had a model where the frequently accessed field are packed close together, to get cache benefit, but that made little difference in practice too.  If we'd like to optimize for cache misses, the overall infrastructure needs to be redone to allow for request array allocations first.  Optimizing the request structure content before we do that is unnecessary.

**Comment by pavanbalaji** [comment]
@raffenet Can you approve this one more time?  Rebased on to `mpich/master`.

**Comment by pavanbalaji** [comment]
Sorry, @raffenet.  Another PR got merged in again.  Can you approve it one more time?

**Pull request merged**